### ISSUE TITLE
es6-transform hang due to wrong acorn dependency is still present in latest published package

### ISSUE THREAD

**Issue by thw0rted** [monounity/karma-typescript#489]
I'm trying to migrate from `karma-webpack` to `karma-typescript` for a large, complex (sadly private) project.  Several project dependencies ship as ES6 modules, so I'm trying to use the es6-transform as laid out in the docs, but once I include it the test build never completes.

The problem is, I tried to put together a [minimal repro](https://github.com/thw0rted/issue-examples/tree/karma-ts-esri) that closely emulates how the larger project is structured, but no matter what I do I can't get it to hang the way the large project does.  In the repro, the debug log output looks like

````text
....
18 08 2021 13:01:59.948:INFO [compiler.karma-typescript]: Compiled 2 files in 1642 ms.
18 08 2021 13:02:00.461:DEBUG [bundler.karma-typescript]: Project has 4 import/require statements, code will be bundled
18 08 2021 13:02:00.481:DEBUG [es6-transform.karma-typescript]: Transforming C:\Workspace\issue-examples\node_modules\@pnp\sp\webs\index.js
18 08 2021 13:02:01.013:DEBUG [es6-transform.karma-typescript]: Transforming C:\Workspace\issue-examples\node_modules\@pnp\sp\rest.js
....
````

It transforms all the JS files in the `@pnp.sp` package then logs "Bundled imports for 4 file(s) in 1900 ms." and gets on with launching the browser and running the tests.  In the problematic project, a very similar test config reaches the first `es6-transform` log line and stops.

Since I can't recreate the issue for you, I'm not sure how to proceed.  I believe I'm already using the most verbose log level available.  Maybe I can hook a debugger to the build process, step into the es6 transform, and see where it's hung up?  I'm not sure how, though.

**Comment by thw0rted** [comment]
I got the debugger working, and I think I've cracked it.

`karma-typescript` and `karma-typescript-angular2-transform` both declare `acorn ^8.x.x` in `dependencies` but `karma-typescript-es6-transform` only declares it in `devDependencies`, even though it's actually used at runtime.  This bug means that `es6-transform` inherited the `acorn` version used by another package in my "big build" (eslint -> espree -> acorn) which happened to be left back at v7.4.1.

Acorn 7.x [required a number value for `ecmaVersion`](https://github.com/acornjs/acorn/tree/7.4.1/acorn#interface) while 8.x added [support for the string "latest"](https://github.com/acornjs/acorn/tree/master/acorn#interface), which is [what you're using](https://github.com/monounity/karma-typescript/blob/master/packages/karma-typescript-es6-transform/src/transform.ts#L62).  Acorn doesn't actually check the value you pass, so if you give it a string, that's what winds up in [this `for` loop](https://github.com/acornjs/acorn/blob/7.4.1/acorn/src/state.js#L15), which spins forever trying to decrement the string.  (Apparently `"hello"--` is `NaN`, then `NaN--` is still `NaN`, so that's fun 😬  )

TL;DR: `es6-transform` needs to declare `acorn` as a non-dev dependency.

**Comment by thw0rted** [comment]
Aaaand you already knew about this, just found #463.  Am I wrong in thinking that you have not cut a new release since then?

**Comment by erikbarke** [comment]
Sorry for the delay, v5.5.2 is on npm now 🚀

**Comment by thw0rted** [comment]
Thanks!  It's taken me most of the week to figure out how to replicate a bunch of stuff I take for granted from Webpack, but I've got most of my tests running and over half of them passing 😁 .

The next hurdle is Angular DI -- everything tells me "can't resolve all parameters".  I'm going to try debugging the angular2-transform. If I can't crack it I'll probably be back here in a couple of hours...
